sdk/typespec/typespec_client_core/src/http/request/mod.rs

```diff
@@ -306,6 +312,225 @@ impl<T> TryFrom<&'static str> for RequestContent<T> {
     }
 }
 
+impl<T> TryFrom<bool> for RequestContent<T> {
+    type Error = Infallible;
+    fn try_from(body: bool) -> Result<Self, Infallible> {
+        Ok(Self {
+            body: Bytes::from(body.to_string()).into(),
+            phantom: PhantomData,
+        })
+    }
+}
+
+#[cfg(feature = "decimal")]
+impl<T> TryFrom<Option<Decimal>> for RequestContent<T> {
+    type Error = Infallible;
+    fn try_from(body: Option<Decimal>) -> Result<Self, Infallible> {
+        Ok(Self {
+            body: Bytes::from(body.map(|d| d.to_string()).unwrap_or_default()).into(),
+            phantom: PhantomData,
+        })
+    }
+}
+
+pub mod json {
```

Comment:
What's the reason to put this in a public child model? It should also be gated on the `json` feature, but in that case you'd do something like:

```rust
#[cfg(feature = "json")]
mod json {
  // ...
}

#[cfg(feature = "json")]
pub use json::*;
```

The module is "private" but exposed publicly as merged. Fewer conditions in source, but I don't see why this needs to be a public submodule.

Reply:
Thank you, I am addressing this in #2833